test/suite/test_compact10.py

```diff
@@ -0,0 +1,212 @@
+
+        # Take the first full backup.
+        os.mkdir(self.backup_dir_1)
+        self.take_full_backup(self.backup_dir_1)
+
+        # Enable background compaction.
+        compact_config = f'background=true,free_space_target=1MB'
+        self.turn_on_bg_compact(compact_config)
+
+        # Wait for all tables to be compacted but the HS.
+        while self.get_files_compacted(uris) < self.num_tables:
+            time.sleep(0.5)
+
+        assert self.get_files_compacted(uris) == self.num_tables
+
+        # Take a second full backup.
+        os.mkdir(self.backup_dir_2)
+        self.take_full_backup(self.backup_dir_2)
+
+        # Compare the backups.
+        for uri in uris:
+            self.compare_backups(uri, self.backup_dir_1, self.backup_dir_2)
+
+        # Background compaction may be have been inspecting a table when disabled which is
+        # considered as an interruption, ignore that message.
+        self.ignoreStdoutPatternIfExists('background compact interrupted by application')
+
+    # This test:
+    # - Creates two full backups before background compaction is enabled, one that will be used for
+    # comparison and another one as a base for incremental backups.
+    # - Performs incremental backups as compaction is rewriting the files until all files have been
+    # compacted.
+    # - Compares the two backups.
+    def test_compact10_incr_backup(self):
+
+        # FIXME-WT-11399
+        if self.runningHook('tiered'):
+            self.skipTest("this test does not yet work with tiered storage")
+
+        # Create and populate tables.
+        uris = []
+        for i in range(self.num_tables):
+            uri = self.uri_prefix + f'_{i}'
+            uris.append(uri)
+            self.session.create(uri, self.create_params)
+            self.populate(uri, self.table_numkv, self.value_size)
+
+        # Write to disk.
+        self.session.checkpoint()
+
+        # Delete 50% of the file.
+        for uri in uris:
+            self.delete_range(uri, 50 * self.table_numkv // 100)
+
+        # Write to disk.
+        self.session.checkpoint()
+
+        # Take two full backups:
+
+        # - The first one will remain untouched.
+        os.mkdir(self.backup_dir_1)
+        self.take_full_backup(self.backup_dir_1)
+
+        # - The second one is used for the incremental backups.
+        os.mkdir(self.backup_dir_2)
+        self.initial_backup = True
+        self.take_full_backup(self.backup_dir_2)
+        self.initial_backup = False
+
+        # Enable background compaction.
+        compact_config = f'background=true,free_space_target=1MB'
+        self.turn_on_bg_compact(compact_config)
+
+        # Every time compaction makes progress, perform an incremental backup.
+        bytes_recovered = 0
+        while self.get_files_compacted(uris) < self.num_tables:
+            new_bytes_recovered = self.get_bytes_recovered()
+            if new_bytes_recovered != bytes_recovered:
```

Comment:
I think it's easier to read this as `if new_bytes_recovered > bytes_recovered`

Reply:
I could agree but we just want to trigger incremental backups, whether we have more bytes or less (not possible?) than the previous check, it is a change.

Reply:
Fair enough, as you say though this stat never decreases.